airflow/lineage/hook.py

```diff
@@ -0,0 +1,149 @@
+
+    It is used to collect the input and output datasets of a hook execution.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.inputs: list[tuple[Dataset, LineageContext]] = []
+        self.outputs: list[tuple[Dataset, LineageContext]] = []
+
+    @staticmethod
+    def create_dataset(dataset_kwargs: dict) -> Dataset:
+        """Create a Dataset instance from the given dataset kwargs."""
+        if "uri" in dataset_kwargs:
+            # Fallback to default factory using the provided URI
+            return Dataset(uri=dataset_kwargs["uri"])
+
+        scheme: str = dataset_kwargs.pop("scheme", None)
+        if not scheme:
+            raise ValueError(
+                "Missing required parameter: either 'uri' or 'scheme' must be provided to create a Dataset."
+            )
+
+        dataset_factory = ProvidersManager().dataset_factories.get(scheme)
+        if not dataset_factory:
+            raise ValueError(
+                f"Unsupported scheme: '{scheme}'. Please provide a valid URI to create a Dataset."
+            )
+
+        return dataset_factory(**dataset_kwargs)
+
+    def add_input_dataset(self, dataset_kwargs: dict, hook: LineageContext) -> None:
+        """Add the input dataset and its corresponding hook execution context to the collector."""
+        try:
+            dataset = self.create_dataset(dataset_kwargs)
+        except Exception as e:
+            self.log.debug("Failed to create AIP-60 compliant Dataset. %s", e)
+            return
+        self.inputs.append((dataset, hook))
+
+    def add_output_dataset(self, dataset_kwargs: dict, hook: LineageContext) -> None:
+        """Add the output dataset and its corresponding hook execution context to the collector."""
+        try:
+            dataset = self.create_dataset(dataset_kwargs)
+        except Exception as e:
+            self.log.debug("Failed to create AIP-60 compliant Dataset. %s", e)
+            return
+        self.outputs.append((dataset, hook))
+
+    @property
+    def collected_datasets(self) -> HookLineage:
+        """Get the collected hook lineage information."""
+        return HookLineage(self.inputs, self.outputs)
+
+    @property
+    def has_collected(self) -> bool:
+        """Check if any datasets have been collected."""
+        return len(self.inputs) != 0 or len(self.outputs) != 0
+
+
+class NoOpCollector(HookLineageCollector):
+    """
+    NoOpCollector is a hook lineage collector that does nothing.
+
+    It is used when you want to disable lineage collection.
+    """
+
+    def add_input_dataset(self, *_):
+        pass
+
+    def add_output_dataset(self, *_):
+        pass
+
+    @property
+    def collected_datasets(
+        self,
+    ) -> HookLineage:
+        self.log.warning(
+            "Data lineage tracking might be incomplete. Consider registering a hook lineage reader for more detailed information."
```

Comment:
Not "might be", right? Maybe something like this:

```suggestion
            "Data lineage tracking is disabled. Register a hook lineage reader to start tracking hook lineage."
```

Reply:
Changed.